contracts/access/RoleChecker.sol

```diff
@@ -0,0 +1,66 @@
+pragma solidity ^0.5.0;
+
+import "./Roles.sol";
+import "../drafts/Counters.sol";
+
+/**
+ * @title RoleChecker
+ * @dev Stores and provides setters and getters for roles using automatically generated ids.
+ * Supports unlimited numbers of roles and addresses.
+ * Role IDs are created with a hash of an integer (autoincremented by Counter) and the contract's address
+ */
+
+contract RoleChecker {
+    using Roles for Roles.Role;
+    using Counters for Counters.Counter;
+
+    mapping (uint256 => Roles.Role) private roles;
+    Counters.Counter private counter;
+
+    event RoleAdded(address indexed account, uint256 roleId);
+    event RoleRemoved(address indexed account, uint256 roleId);
+
+    modifier onlyRole(uint256 _roleId) {
+        require(hasRole(msg.sender, _roleId), "Caller does not have the permissions to call this function");
+        _;
+    }
+
+    function roleExists(uint256 _roleId) internal view returns (bool) { return roles[_roleId].exists; }
```

Comment:
Interesting, I had not considered this! Can you think of use cases where someone might want to check if a role exists?